ocaml/xapi-aux/pool_role.ml

```diff
@@ -80,15 +78,18 @@ let read_pool_role () =
     )
 
 let get_role () =
-  with_pool_role_lock (fun _ ->
-      match !role with
-      | Some x ->
-          x
-      | None ->
-          let r = read_pool_role () in
-          role := Some r ;
-          r
-  )
+  match Atomic.get role with
+  | Some x ->
+      x
+  | None ->
+      with_pool_role_lock (fun _ ->
```

Comment:
Does this suggest that the mutex lock is useless at all? Since it is `Atomic` already, the data will not be corrupted. And meanwhile, it's allowed to read a stale value.